datasets/flwr_datasets/partitioner/vertical_size_partitioner.py

```diff
@@ -0,0 +1,300 @@
+            raise ValueError("partition_sizes must be a list.")
+        if all(isinstance(fraction, float) for fraction in self._partition_sizes):
+            fraction_sum = sum(self._partition_sizes)
+            if fraction_sum != 1.0:
+                raise ValueError("Float ratios in column_distribution must sum to 1.0.")
+            if any(
+                fraction < 0.0 or fraction > 1.0 for fraction in self._partition_sizes
+            ):
+                raise ValueError(
+                    "All floats in column_distribution must be >= 0.0 and <= 1.0."
+                )
+        elif all(
+            isinstance(coulumn_count, int) for coulumn_count in self._partition_sizes
+        ):
+            if any(coulumn_count < 0 for coulumn_count in self._partition_sizes):
+                raise ValueError("All integers in column_distribution must be >= 0.")
+        else:
+            raise ValueError("partition_sizes list must be all floats or all ints.")
+
+        # Validate columns lists
+        for parameter_name, parameter_list in [
+            ("drop_columns", self._drop_columns),
+            ("shared_columns", self._shared_columns),
+            ("active_party_columns", self._active_party_columns),
+        ]:
+            if not all(isinstance(column, str) for column in parameter_list):
+                raise ValueError(f"All entries in {parameter_name} must be strings.")
+
+        valid_modes = {
+            "add_to_first",
+            "add_to_last",
+            "create_as_first",
+            "create_as_last",
+            "add_to_all",
+        }
+        if not (
+            isinstance(self._active_party_columns_mode, int)
+            or self._active_party_columns_mode in valid_modes
+        ):
+            raise ValueError(
+                "active_party_columns_mode must be an int or one of "
+                "'add_to_first', 'add_to_last', 'create_as_first', 'create_as_last', "
+                "'add_to_all'."
+            )
+
+    def _validate_parameters_while_partitioning(
+        self,
+        all_columns: list[str],
+        shared_columns: list[str],
+        active_party_columns: list[str],
+    ) -> None:
+        # Shared columns existance check
+        for column in shared_columns:
+            if column not in all_columns:
+                raise ValueError(f"Shared column '{column}' not found in the dataset.")
+        # Active party columns existence check
+        for column in active_party_columns:
+            if column not in all_columns:
+                raise ValueError(
+                    f"Active party column '{column}' not found in the dataset."
+                )
+        num_columns = len(all_columns)
+        if all(isinstance(size, int) for size in self._partition_sizes):
+            if sum(self._partition_sizes) > num_columns:
+                raise ValueError(
+                    "Sum of partition sizes cannot exceed the total number of columns."
```

Comment:
Should this check be modified so it tests that the `sum` is exactly the number of columns? (so to match the logic done when the `partition_sizes` are passed as `list[float]` ? (`_validate_parameters_in_init()` checks if they add up to `1.0`)

Reply:
Thanks, yeah, let's keep it consistent. I'll change it.

Reply:
Done